HUD example compiles but crashes

This crashes as soon as I launch the example with the following traceback:

```bash
fatal error: unexpected signal during runtime execution
[signal 0xb code=0x1 addr=0x0 pc=0x7f90bf226ff0]

runtime stack:
runtime.throw(0x7bac40, 0x2a)
	/home/faide/tools/go/src/runtime/panic.go:527 +0x90
runtime.sigpanic()
	/home/faide/tools/go/src/runtime/sigpanic_unix.go:12 +0x5a

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x60d8e0, 0xc8200c7ad0, 0x0)
	/home/faide/tools/go/src/runtime/cgocall.go:120 +0x11b fp=0xc8200c7a88 sp=0xc8200c7a58
github.com/go-gl/gl/v2.1/gl._Cfunc_glowDrawElements(0x7f90bef73760, 0x600000004, 0x1403, 0x0)
	github.com/go-gl/gl/v2.1/gl/_obj/_cgo_gotypes.go:12227 +0x35 fp=0xc8200c7ad0 sp=0xc8200c7a88
github.com/go-gl/gl/v2.1/gl.DrawElements(0x600000004, 0xc800001403, 0x0)
	/home/faide/go/src/github.com/go-gl/gl/v2.1/gl/package.go:19385 +0x45 fp=0xc8200c7af8 sp=0xc8200c7ad0
github.com/paked/webgl.(*Context).DrawElements(0xc8200f6000, 0x4, 0x6, 0x1403, 0x0)
	/home/faide/go/src/github.com/paked/webgl/webgl_gl2.go:775 +0x42 fp=0xc8200c7b18 sp=0xc8200c7af8
github.com/paked/engi.(*HUDShader).flush(0xc8200be000)
	/home/faide/go/src/github.com/paked/engi/shaders.go:289 +0x168 fp=0xc8200c7b58 sp=0xc8200c7b18
github.com/paked/engi.(*HUDShader).Post(0xc8200be000)
	/home/faide/go/src/github.com/paked/engi/shaders.go:294 +0x21 fp=0xc8200c7b68 sp=0xc8200c7b58
github.com/paked/engi.(*RenderSystem).Post(0xc82000e420)
	/home/faide/go/src/github.com/paked/engi/render.go:291 +0xce fp=0xc8200c7ca0 sp=0xc8200c7b68
github.com/paked/engi/ecs.(*World).Update(0xc820014720, 0x0)
	/home/faide/go/src/github.com/paked/engi/ecs/world.go:121 +0x2d5 fp=0xc8200c7d90 sp=0xc8200c7ca0
github.com/paked/engi.RunIteration()
	/home/faide/go/src/github.com/paked/engi/engi_glfw.go:194 +0x53 fp=0xc8200c7da8 sp=0xc8200c7d90
github.com/paked/engi.runLoop(0x7f90bc69f360, 0xc6fb60, 0x400)
	/home/faide/go/src/github.com/paked/engi/engi_glfw.go:226 +0xd9 fp=0xc8200c7e70 sp=0xc8200c7da8
github.com/paked/engi.Open(0x0, 0x71d4b0, 0x8, 0x0, 0x400, 0x280, 0x0, 0x0, 0x7f90bc69f360, 0xc6fb60)
	/home/faide/go/src/github.com/paked/engi/engi.go:68 +0xf4 fp=0xc8200c7ea0 sp=0xc8200c7e70
main.main()
	/home/faide/go/src/github.com/paked/engi/demos/hud/hud.go:108 +0xab fp=0xc8200c7f50 sp=0xc8200c7ea0
runtime.main()
	/home/faide/tools/go/src/runtime/proc.go:111 +0x2b0 fp=0xc8200c7fa0 sp=0xc8200c7f50
runtime.goexit()
	/home/faide/tools/go/src/runtime/asm_amd64.s:1721 +0x1 fp=0xc8200c7fa8 sp=0xc8200c7fa0

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/home/faide/tools/go/src/runtime/asm_amd64.s:1721 +0x1
```

if I suppress the call to ```w.AddEntity(hudBg)``` it works so I suspect something in ```generateHUDBackground``` but I can't see what really differs from ```generateBackground``` which works fine.

If I try with go 1.6 I get
```bash
panic: runtime error: cgo argument has Go pointer to Go pointer

goroutine 1 [running, locked to thread]:
panic(0x6d7c60, 0xc82000a610)
	/home/faide/tools/go/src/runtime/panic.go:464 +0x3e6
github.com/go-gl/gl/v2.1/gl.ShaderSource(0x100000001, 0xc8200340c0, 0x0)
	/home/faide/go/src/github.com/go-gl/gl/v2.1/gl/package.go:24310 +0x4c
github.com/paked/webgl.(*Context).ShaderSource(0xc8200f4000, 0xc82000a5d8, 0x96aba0, 0x1e9)
	/home/faide/go/src/github.com/paked/webgl/webgl_gl2.go:620 +0xc3
github.com/paked/engi.LoadShader(0x96aba0, 0x1e9, 0x96a3e0, 0x115, 0x0)
	/home/faide/go/src/github.com/paked/engi/assets.go:136 +0x76
github.com/paked/engi.(*DefaultShader).Initialize(0xc47268, 0x4420000044800000)
	/home/faide/go/src/github.com/paked/engi/shaders.go:73 +0x45
github.com/paked/engi.(*RenderSystem).New(0xc82000e480, 0xc820014690)
	/home/faide/go/src/github.com/paked/engi/render.go:209 +0x1a3
github.com/paked/engi/ecs.(*World).AddSystem(0xc820014690, 0x7f99024b6450, 0xc82000e480)
	/home/faide/go/src/github.com/paked/engi/ecs/world.go:65 +0x35
main.(*Game).Setup(0xc62bf8, 0xc820014690)
	/home/faide/go/src/github.com/paked/engi/demos/hud/hud.go:78 +0x87
github.com/paked/engi.SetScene(0x7f99024b6360, 0xc62bf8, 0xc820014600)
	/home/faide/go/src/github.com/paked/engi/scene.go:86 +0x2a7
github.com/paked/engi.RunPreparation(0x7f99024b6360, 0xc62bf8)
	/home/faide/go/src/github.com/paked/engi/engi_glfw.go:215 +0x30f
github.com/paked/engi.runLoop(0x7f99024b6360, 0xc62bf8, 0x400)
	/home/faide/go/src/github.com/paked/engi/engi_glfw.go:219 +0x3d
github.com/paked/engi.Open(0x0, 0x71ad00, 0x8, 0x0, 0x400, 0x280, 0x0, 0x0, 0x7f99024b6360, 0xc62bf8)
	/home/faide/go/src/github.com/paked/engi/engi.go:68 +0xf4
main.main()
	/home/faide/go/src/github.com/paked/engi/demos/hud/hud.go:108 +0xac
```

--- a/font.go
+++ b/font.go
@@ -119,7 +119,7 @@ func (f *Font) Render(text string) *Texture {
 	c.SetSrc(fg)
 
 	// Draw the text.
-	pt := freetype.Pt(0, int(yBearing))
+	pt := fixed.P(0, int(yBearing))
 	_, err := c.DrawString(text, pt)
 	if err != nil {
 		log.Println(err)

--- a/shaders.go
+++ b/shaders.go
@@ -147,19 +147,13 @@ type HUDShader struct {
 	projX float32
 	projY float32
 
-	lastTexture        *webgl.Texture
-	drawCount          int
-	coordBuffer        *webgl.Buffer
-	coordBufferContent []float32
-	coordIndices       []uint16
-	coordIndicesVBO    *webgl.Buffer
+	lastTexture *webgl.Texture
 
-	inPosition    int
-	inTexCoords   int
-	inColor       int
-	inCoordinates int
-	ufPosition    *webgl.UniformLocation
-	ufProjection  *webgl.UniformLocation
+	inPosition   int
+	inTexCoords  int
+	inColor      int
+	ufPosition   *webgl.UniformLocation
+	ufProjection *webgl.UniformLocation
 }
 
 func (s *HUDShader) Initialize(width, height float32) {
@@ -169,8 +163,8 @@ func (s *HUDShader) Initialize(width, height float32) {
 attribute vec2 in_Position;
 attribute vec2 in_TexCoords;
 attribute vec4 in_Color;
-attribute vec2 in_Coordinates;
 
+uniform vec2 uf_Position;
 uniform vec2 uf_Projection;
 
 varying vec4 var_Color;
@@ -180,8 +174,8 @@ void main() {
   var_Color = in_Color;
   var_TexCoords = in_TexCoords;
 
-  gl_Position = vec4((in_Position.x + in_Coordinates.x)/  uf_Projection.x - 1.0,
-  					 (in_Position.y + in_Coordinates.y)/ -uf_Projection.y + 1.0,
+  gl_Position = vec4((in_Position.x + uf_Position.x)/  uf_Projection.x - 1.0,
+  					 (in_Position.y + uf_Position.y)/ -uf_Projection.y + 1.0,
   					 0.0, 1.0);
 
 }`, `
@@ -202,19 +196,6 @@ void main (void) {
 }`)
 
 	// Create and populate indices buffer
-	s.coordIndices = make([]uint16, 6*bufferSize)
-	for i, j := 0, 0; i < bufferSize*6; i, j = i+6, j+4 {
-		s.coordIndices[i+0] = uint16(j + 0)
-		s.coordIndices[i+1] = uint16(j + 0)
-		s.coordIndices[i+2] = uint16(j + 0)
-		s.coordIndices[i+3] = uint16(j + 0)
-		s.coordIndices[i+4] = uint16(j + 0)
-		s.coordIndices[i+5] = uint16(j + 0)
-	}
-	s.coordIndicesVBO = Gl.CreateBuffer()
-	Gl.BindBuffer(Gl.ELEMENT_ARRAY_BUFFER, s.coordIndicesVBO)
-	Gl.BufferData(Gl.ELEMENT_ARRAY_BUFFER, s.coordIndices, Gl.STATIC_DRAW)
-
 	s.indices = make([]uint16, 6*bufferSize)
 	for i, j := 0, 0; i < bufferSize*6; i, j = i+6, j+4 {
 		s.indices[i+0] = uint16(j + 0)
@@ -234,7 +215,6 @@ void main (void) {
 	s.inPosition = Gl.GetAttribLocation(s.program, "in_Position")
 	s.inTexCoords = Gl.GetAttribLocation(s.program, "in_TexCoords")
 	s.inColor = Gl.GetAttribLocation(s.program, "in_Color")
-	s.inCoordinates = Gl.GetAttribLocation(s.program, "in_Coordinates")
 
 	// Define things that should be set per draw
 	s.ufPosition = Gl.GetUniformLocation(s.program, "uf_Position")
@@ -244,7 +224,6 @@ void main (void) {
 	Gl.EnableVertexAttribArray(s.inPosition)
 	Gl.EnableVertexAttribArray(s.inTexCoords)
 	Gl.EnableVertexAttribArray(s.inColor)
-	Gl.EnableVertexAttribArray(s.inCoordinates)
 
 	Gl.Enable(Gl.BLEND)
 	Gl.BlendFunc(Gl.SRC_ALPHA, Gl.ONE_MINUS_SRC_ALPHA)
@@ -254,11 +233,7 @@ void main (void) {
 
 func (s *HUDShader) Pre() {
 	Gl.UseProgram(s.program)
-	Gl.BindBuffer(Gl.ELEMENT_ARRAY_BUFFER, s.indexVBO)
-
 	Gl.Uniform2f(s.ufProjection, s.projX, s.projY)
-
-	s.coordBufferContent = make([]float32, 0)
 }
 
 func (s *HUDShader) Draw(texture *webgl.Texture, buffer *webgl.Buffer, x, y, rotation float32) {
@@ -273,25 +248,11 @@ func (s *HUDShader) Draw(texture *webgl.Texture, buffer *webgl.Buffer, x, y, rot
 		s.lastTexture = texture
 	}
 
-	s.coordBufferContent = append(s.coordBufferContent, []float32{x, y, rotation}...)
-	s.drawCount++
-
-	if s.drawCount > bufferSize {
-		s.flush()
-	}
-}
-
-func (s *HUDShader) flush() {
-	Gl.BindBuffer(Gl.ARRAY_BUFFER, s.coordBuffer)
-	Gl.BufferData(Gl.ARRAY_BUFFER, s.coordBufferContent, Gl.DYNAMIC_DRAW)
-	Gl.VertexAttribPointer(s.inCoordinates, 3, Gl.FLOAT, false, 12, 0)
-
-	Gl.DrawElements(Gl.TRIANGLES, 6*s.drawCount, Gl.UNSIGNED_SHORT, 0)
-	s.drawCount = 0
+	Gl.Uniform2f(s.ufPosition, x, y)
+	Gl.DrawElements(Gl.TRIANGLES, 6, Gl.UNSIGNED_SHORT, 0)
 }
 
 func (s *HUDShader) Post() {
-	s.flush()
 	s.lastTexture = nil
 }
 

